helix-view/src/document.rs

```diff
@@ -979,6 +1000,7 @@ impl Document {
             rev
         );
         self.last_saved_revision = rev;
+        self.last_saved_time = SystemTime::now();
```

Comment:
Since the file isn't saved, I don't think the last_saved_time should be set.

Reply:
My thinking here is to mimic the way History handles revisions and last_saved_revision here. They have an initial value as if the file had been saved at the time it was opened, this avoids adding a lot of unnecessary handling for an Option<>.